### DISCUSSION TITLE
There are 869 problems when you open antlr project in vscode

### DISCUSSION THREAD

**Discussion by chris-miner** [antlr/antlr4#3911] [General]
I can build the project from the command line, and run at least the java tests.  So there aren't really 869 problems.  However, it is hard to see actual problems vs reported problems because there are 869 of them reported.

I've looked into a few.  Mostly it is things like, one path or another is unexpected.  Or a path doesn't exist.  Or configuration is non-standard.  Without an IDE, I suppose I'd never see any of them.

What are people seeing in IntelliJ?  Or Eclipse?  VSCode?

**Comment by jimidle** [comment]
IntelliJ gives this, if you run the inspection system against all code:

![image](https://user-images.githubusercontent.com/1309104/192190612-d22e160b-e5ac-4fe5-81db-a7a3a793d418.png)

Most are warnings. Java in a bit more detail:

![image](https://user-images.githubusercontent.com/1309104/192190801-5bfe68a4-362a-46f1-aee0-7b6ad989e86d.png)

**Reply by chris-miner** [reply]
Okay, that doesn't look too bad, especially when you have to run the inspection system.  Those look like a lot of warnings.   In vscode I'm seeing a lot of errors (528).  A bunch are duplicates — for example "ANTLRParser cannot be resolved to a variable".  Filtering out duplicates based on the message yields only 148 errors.  Among those there are 35 errors about vscode not being able to resolve a type or variable.

Ignoring the message clutter for the moment, these errors mean I can't run many of the individual tests via the IDE or step through them in the debugger because vscode complains the build failed.

Maybe I should just create a branch in my fork to contain the patches to remedy the errors at least.  Then I can merge those as needed to updates on the dev and main branches.  Something like keep a branch for build issues, leave dev/main untouched.  Branch dev/main to do work.  Then before doing the work, merge in changes from my patch branch.

Edit: Now that I think about it, that makes it hard to do PRs.  My PR (work) branch will include my build patches.
